### ISSUE TITLE
Unable to deeplink due to 400 on https://api.branch.io/v1/open

### ISSUE THREAD

**Issue by philipengberg** [BranchMetrics/ios-branch-deep-linking#823]
*By request from your support team, I hereby create an issue on this repo.*

We're seeing a lot of iOS devices, that refuse to deeplink from any Branch link. After debugging it we found that the Branch SDK produced this error on those devices: 

`BNCServerInterface.m(598) Error: An error prevented request to https://api.branch.io/v1/open from completing: Error Domain=io.branch.sdk.error Code=1003 "The network request was invalid."`

You support team then asked to get full logs, which I'll provide here:

### On app startup we get:

```
2018-04-18 09:59:00.691382+0200 Tonsser[21498:3068036] [branch.io] Branch.m(192) Log: Branch version 0.24.1 started at 2018-04-18 07:59:00 +0000.
2018-04-18 09:59:00.699768+0200 Tonsser[21498:3068043] [branch.io] BNCServerInterface.m(759) Debug: URL: https://api.branch.io/v1/open.
2018-04-18 09:59:00.700032+0200 Tonsser[21498:3068043] [branch.io] BNCServerInterface.m(763) Debug: Body: {
    "branch_key" = [redacted];
hne?link_click_id=485466565792233069","local_ip":"10.134.110.132","brand":"Apple","lastest_update_time":1524038321000,"update":2,"cd":{"pn":"com.tonsser.tonsser","mv":"-1"},"screen_height":2436,"apple_ad_attribution_checked":false,"model":"iPhone10,6","branch_key":[redacted],"user_agent":"Mozilla/5.0 (iPhone; CPU iPhone OS 11_3 like Mac OS X) AppleWebKit/605.1.15 (KHTML, like Gecko) Mobile/15E216","first_install_time":1511011032163,"hardware_id":[redacted]
2018-04-18 09:59:00.700752+0200 Tonsser[21498:3068043] [branch.io] BNCNetworkService.m(232) Debug: Network start operation https://api.branch.io/v1/open.
2018-04-18 09:59:01.096089+0200 Tonsser[21498:3068067] [branch.io] BNCNetworkService.m(227) Debug: Network finish operation https://api.branch.io/v1/open 0.396s. Status 400 error (null).
2018-04-18 09:59:01.096259+0200 Tonsser[21498:3068067] [branch.io] BNCServerInterface.m(819) Debug: Server returned: Status: 400; Data: {
2018-04-18 09:59:01.096387+0200 Tonsser[21498:3068067] [branch.io] BNCServerInterface.m(598) Error: An error prevented request to https://api.branch.io/v1/open from completing: Error Domain=io.branch.sdk.error Code=1003 "The network request was invalid." UserInfo={NSLocalizedDescription=2018-04-18 09:59:01.503923+0200 Tonsser[21498:3068044] [branch.io] BNCServerInterface.m(759) Debug: URL: https://api.branch.io/v1/open.
2018-04-18 09:59:01.504220+0200 Tonsser[21498:3068035] [branch.io] BNCServerInterface.m(763) Debug: Body: {
    "branch_key" = [redacted];
92233069","local_ip":"10.134.110.132","brand":"Apple","lastest_update_time":1524038321000,"update":2,"cd":{"pn":"com.tonsser.tonsser","mv":"-1"},"screen_height":2436,"apple_ad_attribution_checked":false,"model":"iPhone10,6","branch_key":[redacted],"user_agent":"Mozilla/5.0 (iPhone; CPU iPhone OS 11_3 like Mac OS X) AppleWebKit/605.1.15 (KHTML, like Gecko) Mobile/15E216","first_install_time":1511011032163,"hardware_id":[redacted],"os":"iOS","identity_id":"411130905558697046","hardware_id_type":"random","latest_install_time":1511011032163,"screen_width":1125,"language":"en","ios_vendor_id":"55975D11-B4F9-41E8-9A5E-EA7C5FC026D0","app_version":"6.0.6"}.
2018-04-18 09:59:01.506009+0200 Tonsser[21498:3068035] [branch.io] BNCNetworkService.m(232) Debug: Network start operation https://api.branch.io/v1/open.
2018-04-18 09:59:01.694457+0200 Tonsser[21498:3068041] [branch.io] BNCNetworkService.m(227) Debug: Network finish operation https://api.branch.io/v1/open 0.190s. Status 400 error (null).
2018-04-18 09:59:01.694510+0200 Tonsser[21498:3068041] [branch.io] BNCServerInterface.m(819) Debug: Server returned: Status: 400; Data: {
2018-04-18 09:59:01.694587+0200 Tonsser[21498:3068041] [branch.io] BNCServerInterface.m(598) Error: An error prevented request to https://api.branch.io/v1/open from completing: Error Domain=io.branch.sdk.error Code=1003 "The network request was invalid." UserInfo={NSLocalizedDescription=The network request was invalid., NSLocalizedFailureReason=The request was invalid.}
```

### On Branch link tapped

```
2018-04-18 09:59:53.429898+0200 Tonsser[21498:3068525] [branch.io] BNCServerInterface.m(759) Debug: URL: https://api.branch.io/v1/open.
2018-04-18 09:59:53.430645+0200 Tonsser[21498:3068793] [branch.io] BNCServerInterface.m(763) Debug: Body: {
    "branch_key" = [redacted];
t_id":"411781086872536794","uri_scheme":"ak237526079791417","retryNumber":0,"previous_update_time":1523569193000,"universal_link_url":"^fb\d+:","facebook_app_link_checked":false,"external_intent_uri":"com.tonsser.tonsser://tonsser.com/users/jack-lahne?link_click_id=485466565792233069","local_ip":"10.134.110.132","brand":"Apple","lastest_update_time":1524038321000,"update":2,"cd":{"pn":"com.tonsser.tonsser","mv":"-1"},"screen_height":2436,"apple_ad_attribution_checked":false,"model":"iPhone10,6","branch_key"
2018-04-18 09:59:53.433991+0200 Tonsser[21498:3068793] [branch.io] BNCNetworkService.m(232) Debug: Network start operation https://api.branch.io/v1/open.
2018-04-18 09:59:53.646281+0200 Tonsser[21498:3068304] [branch.io] BNCNetworkService.m(227) Debug: Network finish operation https://api.branch.io/v1/open 0.216s. Status 400 error (null).
2018-04-18 09:59:53.646363+0200 Tonsser[21498:3068304] [branch.io] BNCServerInterface.m(819) Debug: Server returned: Status: 400; Data: {
2018-04-18 09:59:53.646408+0200 Tonsser[21498:3068304] [branch.io] BNCServerInterface.m(598) Error: An error prevented request to https://api.branch.io/v1/open from completing: Error Domain=io.branch.sdk.error Code=1003 "The network request was invalid." UserInfo={NSLocalizedDescription=The network request was invalid., NSLocalizedFailureReason=The request was invalid.}
2018-04-18 09:59:54.180317+0200 Tonsser[21498:3068793] [branch.io] BNCServerInterface.m(759) Debug: URL: https://api.branch.io/v1/open.
2018-04-18 09:59:54.181636+0200 Tonsser[21498:3068793] [branch.io] BNCServerInterface.m(763) Debug: Body: {
    "branch_key" = [redacted];
t_id":"411781086872536794","uri_scheme":"ak237526079791417","retryNumber":0,"previous_update_time":1523569193000,"universal_link_url":"^fb\d+:","facebook_app_link_checked":false,"external_intent_uri":"com.tonsser.tonsser://tonsser.com/users/jack-lahne?link_click_id=485466565792233069","local_ip":"10.134.110.132","brand":"Apple","lastest_update_time":1524038321000,"update":2,"cd":{"pn":"com.tonsser.tonsser","mv":"-1"},"screen_height":2436,"apple_ad_attribution_checked":false,"model":"iPhone10,6","branch_key"
2018-04-18 09:59:54.206659+0200 Tonsser[21498:3068793] [branch.io] BNCNetworkService.m(232) Debug: Network start operation https://api.branch.io/v1/open.
2018-04-18 09:59:54.377973+0200 Tonsser[21498:3068527] [branch.io] BNCNetworkService.m(227) Debug: Network finish operation https://api.branch.io/v1/open 0.197s. Status 400 error (null).
2018-04-18 09:59:54.378022+0200 Tonsser[21498:3068527] [branch.io] BNCServerInterface.m(819) Debug: Server returned: Status: 400; Data: {
2018-04-18 09:59:54.378098+0200 Tonsser[21498:3068527] [branch.io] BNCServerInterface.m(598) Error: An error prevented request to https://api.branch.io/v1/open from completing: Error Domain=io.branch.sdk.error Code=1003 "The network request was invalid." UserInfo={NSLocalizedDescription=The network request was invalid., NSLocalizedFailureReason=The request was invalid.}
```

As mentioned it only happens to some users and we have found no indicators as to why those users are affected. Force closing the app does not help.

**Comment by E-B-Smith** [comment]
Thanks for reporting this.  I'll look into it and see what I find.

**Comment by nextbigidea** [comment]
Having same issue....

`2018-04-19 17:40:58.553035-0400 VICI[8508:3031715] [branch.io] Branch.m(192) Log: Branch version 0.24.1 started at 2018-04-19 21:40:58 +0000.
2018-04-19 17:40:58.786011-0400 VICI[8508:3031719] [branch.io] BNCServerInterface.m(598) Error: An error prevented request to https://api.branch.io/v1/open from completing: Error Domain=io.branch.sdk.error Code=1003 "The network request was invalid." UserInfo={NSLocalizedDescription=The network request was invalid., NSLocalizedFailureReason=The request was invalid.}
2018-04-19 17:46:05.996035-0400 VICI[8508:3034291] [branch.io] BNCServerInterface.m(598) Error: An error prevented request to https://api.branch.io/v1/open from completing: Error Domain=io.branch.sdk.error Code=1003 "The network request was invalid." UserInfo={NSLocalizedDescription=The network request was invalid., NSLocalizedFailureReason=The request was invalid.}`

**Comment by nextbigidea** [comment]
My app now no longer shows up on branch dashboard either...

**Comment by E-B-Smith** [comment]
@nextbigidea Sorry about your issue!  

Can you turn logging on and send me the logs?
To turn Branch logging on in the app, add the line
```
BNCLogSetDisplayLevel(BNCLogLevelAll);
```
in your `application:didFinishLaunchingWithOptions:` method.

**Comment by E-B-Smith** [comment]
@nextbigidea email me at esmith@branch.io.

**Comment by E-B-Smith** [comment]
Try making sure your Branch key is correct and not truncated.

**Comment by nextbigidea** [comment]
@E-B-Smith it's not truncated

**Comment by E-B-Smith** [comment]
Thank you both for all the info.

I have a fix in PR https://github.com/BranchMetrics/ios-branch-deep-linking/pull/824 which I'll promptly release after some more testing.

**Comment by E-B-Smith** [comment]
Fixed in release 0.24.2.
